src/components/common/note-loading-boundary/note-loading-boundary.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { PropsWithChildren } from 'react'
+import React, { Fragment } from 'react'
+import { useLoadNoteFromServer } from './hooks/use-load-note-from-server'
+import { LoadingScreen } from '../../application-loader/loading-screen/loading-screen'
+import { CommonErrorPage } from '../../error-pages/common-error-page'
+
+/**
+ * Loads the note identified by the note-id in the URL.
+ * During the loading a {@link LoadingScreen loading screen} will be rendered instead of the child elements.
+ * The boundary also shows errors that occur during the loading process.
+ *
+ * @param children The react elements that will be shown when the loading was successful.
+ */
+export const NoteLoadingBoundary: React.FC<PropsWithChildren<unknown>> = ({ children }) => {
+  const { error, loading } = useLoadNoteFromServer()
+
+  if (loading) {
+    return <LoadingScreen />
+  } else if (error) {
+    return (
+      <CommonErrorPage
+        titleI18nKey={'noteLoadingBoundary.errorWhileLoadingContent'}
+        descriptionI18nKey={error.message}></CommonErrorPage>
```

Comment:
Is there anything prohibiting this?

```suggestion
        descriptionI18nKey={error.message} />
```